### ISSUE TITLE
baritone crashes game on world load

### ISSUE THREAD

**Issue by RoguePortal** [ImpactDevelopment/ImpactClient#1092]
Game crashes on load of world/server. This occurred after using auto mine/legit mine. Happens 100% time now, immediately after load.

Minecraft launcher error:

Error: java.lang.NullPointerException: Ticking entity

Crash report:

<details><summary><strong>Crash report</strong></summary>

```
---- Minecraft Crash Report ----
// Oops.

Time: 7/04/19 6:21 PM
Description: Ticking entity

java.lang.NullPointerException: Ticking entity
	at java.util.Objects.requireNonNull(Objects.java:203)
	at baritone.p.a(Unknown Source)
	at baritone.ch$$Lambda$3456/1054571493.apply(Unknown Source)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:512)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:502)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:576)
	at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:255)
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:438)
	at baritone.ch.mineByName(Unknown Source)
	at baritone.api.process.IMineProcess.mineByName(Unknown Source)
	at me.zero.clarinet.3I.lambda$new$1(3I.java:10)
	at me.zero.clarinet.3I$$Lambda$1600/1391691162.invoke(Unknown Source)
	at me.zero.alpine.listener.Listener.invoke(Listener.java:93)
	at me.zero.alpine.bus.EventManager.lambda$post$5(EventManager.java:99)
	at me.zero.alpine.bus.EventManager$$Lambda$1087/28635717.accept(Unknown Source)
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:890)
	at me.zero.alpine.bus.EventManager.post(EventManager.java:99)
	at clientapi.7.post(7.java:7)
	at ctj.handler$onPostUpdate$zzd000(SourceFile:1221)
	at ctj.R_(SourceFile:234)
	at axy.a(SourceFile:1243)
	at axy.g(SourceFile:1218)
	at axy.o_(SourceFile:1100)
	at cft.m(SourceFile:1281)
	at cft.c(SourceFile:821)
	at cft.a(SourceFile:395)
	at net.minecraft.client.main.Main.main(SourceFile:144)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at java.util.Objects.requireNonNull(Objects.java:203)
	at baritone.p.a(Unknown Source)
	at baritone.ch$$Lambda$3456/1054571493.apply(Unknown Source)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:512)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:502)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:576)
	at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:255)
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:438)
	at baritone.ch.mineByName(Unknown Source)
	at baritone.api.process.IMineProcess.mineByName(Unknown Source)
	at me.zero.clarinet.3I.lambda$new$1(3I.java:10)
	at me.zero.clarinet.3I$$Lambda$1600/1391691162.invoke(Unknown Source)
	at me.zero.alpine.listener.Listener.invoke(Listener.java:93)
	at me.zero.alpine.bus.EventManager.lambda$post$5(EventManager.java:99)
	at me.zero.alpine.bus.EventManager$$Lambda$1087/28635717.accept(Unknown Source)
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:890)
	at me.zero.alpine.bus.EventManager.post(EventManager.java:99)
	at clientapi.7.post(7.java:7)
	at ctj.handler$onPostUpdate$zzd000(SourceFile:1221)
	at ctj.R_(SourceFile:234)
	at axy.a(SourceFile:1243)
	at axy.g(SourceFile:1218)

-- Entity being ticked --
Details:
	Entity Type: minecraft:player (ctj)
	Entity ID: 590
	Entity Name: RoguePortal
	Entity's Exact location: 310.47, 94.00, -203.44
	Entity's Block location: World: (310,94,-204), Chunk: (at 6,5,4 in 19,-13; contains blocks 304,0,-208 to 319,255,-193), Region: (0,-1; contains chunks 0,-32 to 31,-1, blocks 0,0,-512 to 511,255,-1)
	Entity's Momentum: 0.00, -0.08, 0.00
	Entity's Passengers: []
	Entity's Vehicle: ~~ERROR~~ NullPointerException: null
Stacktrace:
	at axy.o_(SourceFile:1100)

-- Affected level --
Details:
	Level name: MpServer
	All players: 1 total; [ctj['RoguePortal'/590, l='MpServer', x=310.47, y=94.00, z=-203.44]]
	Chunk stats: MultiplayerChunkCache: 625, 625
	Level seed: 0
	Level generator: ID 00 - default, ver 1. Features enabled: false
	Level generator options: {}
	Level spawn location: World: (240,64,-64), Chunk: (at 0,4,0 in 15,-4; contains blocks 240,0,-64 to 255,255,-49), Region: (0,-1; contains chunks 0,-32 to 31,-1, blocks 0,0,-512 to 511,255,-1)
	Level time: 181167 game time, 172678 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
	Forced entities: 99 total; [ajl['Bat'/32, l='MpServer', x=234.10, y=24.06, z=-249.68], anq['Spider'/35, l='MpServer', x=231.02, y=16.00, z=-220.70], akf['Sheep'/36, l='MpServer', x=231.31, y=67.00, z=-212.03], ajt['Cow'/37, l='MpServer', x=237.12, y=63.00, z=-160.52], ajt['Cow'/40, l='MpServer', x=252.48, y=63.00, z=-150.82], ajr['Chicken'/46, l='MpServer', x=267.48, y=89.00, z=-238.15], amm['Egg'/47, l='MpServer', x=268.28, y=89.00, z=-238.11], ajt['Cow'/48, l='MpServer', x=258.31, y=67.00, z=-205.80], amm['Rotten Flesh'/49, l='MpServer', x=264.33, y=66.00, z=-204.10], ajt['Cow'/50, l='MpServer', x=260.83, y=63.00, z=-165.50], ajl['Bat'/58, l='MpServer', x=283.20, y=14.46, z=-149.98], ajl['Bat'/59, l='MpServer', x=271.38, y=26.45, z=-126.51], ajl['Bat'/60, l='MpServer', x=271.86, y=29.15, z=-124.78], ajl['Bat'/316, l='MpServer', x=385.86, y=19.29, z=-247.02], ann['Skeleton'/64, l='MpServer', x=302.50, y=59.00, z=-228.50], ann['Skeleton'/65, l='MpServer', x=299.50, y=59.00, z=-225.50], akd['Rabbit'/321, l='MpServer', x=389.33, y=82.00, z=-201.22], ann['Skeleton'/66, l='MpServer', x=302.50, y=59.00, z=-230.50], akf['Sheep'/322, l='MpServer', x=387.23, y=82.00, z=-197.48], anw['Zombie'/67, l='MpServer', x=294.50, y=77.00, z=-229.50], ann['Skeleton'/68, l='MpServer', x=298.51, y=62.00, z=-220.73], anw['Zombie'/69, l='MpServer', x=292.76, y=27.00, z=-153.50], anw['Zombie'/70, l='MpServer', x=289.61, y=46.00, z=-135.50], anw['Zombie'/71, l='MpServer', x=292.50, y=45.67, z=-136.44], akf['Sheep'/74, l='MpServer', x=307.07, y=85.00, z=-237.69], ajt['Cow'/75, l='MpServer', x=308.17, y=85.00, z=-233.26], ajt['Cow'/76, l='MpServer', x=308.03, y=85.00, z=-238.01], ajt['Cow'/77, l='MpServer', x=308.15, y=85.00, z=-238.93], ajt['Cow'/78, l='MpServer', x=307.07, y=85.00, z=-238.93], ajt['Cow'/79, l='MpServer', x=312.93, y=85.00, z=-238.93], ajt['Cow'/80, l='MpServer', x=307.07, y=85.00, z=-236.79], akf['Sheep'/81, l='MpServer', x=312.93, y=85.00, z=-236.21], akf['Sheep'/82, l='MpServer', x=312.30, y=85.00, z=-237.85], akf['Sheep'/83, l='MpServer', x=309.27, y=85.00, z=-238.93], akf['Sheep'/84, l='MpServer', x=310.60, y=85.00, z=-235.42], akf['Sheep'/85, l='MpServer', x=309.95, y=85.00, z=-236.36], akf['Sheep'/86, l='MpServer', x=308.08, y=85.00, z=-234.21], akf['Sheep'/87, l='MpServer', x=311.94, y=85.00, z=-234.74], ajt['Cow'/88, l='MpServer', x=307.61, y=85.00, z=-235.87], ajt['Cow'/89, l='MpServer', x=312.09, y=85.00, z=-238.79], ajt['Cow'/90, l='MpServer', x=308.06, y=85.00, z=-236.35], akf['Sheep'/91, l='MpServer', x=310.31, y=85.00, z=-238.82], akf['Sheep'/92, l='MpServer', x=311.23, y=85.00, z=-238.93], anw['Zombie'/95, l='MpServer', x=321.50, y=54.00, z=-243.50], anw['Zombie'/96, l='MpServer', x=322.50, y=37.00, z=-196.50], ann['Skeleton'/97, l='MpServer', x=324.50, y=37.00, z=-197.50], ann['Skeleton'/98, l='MpServer', x=324.50, y=23.00, z=-181.50], amm['Spruce Sapling'/99, l='MpServer', x=322.36, y=84.00, z=-173.70], amt['Creeper'/100, l='MpServer', x=330.50, y=82.00, z=-162.50], anw['Zombie'/101, l='MpServer', x=328.50, y=82.00, z=-162.50], apl['Minecart with Chest'/108, l='MpServer', x=350.50, y=17.00, z=-214.89], amm['Rotten Flesh'/109, l='MpServer', x=350.98, y=87.00, z=-206.87], ajl['Bat'/110, l='MpServer', x=347.95, y=13.32, z=-185.51], amt['Creeper'/111, l='MpServer', x=339.50, y=23.00, z=-177.50], ajl['Bat'/112, l='MpServer', x=334.88, y=23.04, z=-176.25], anw['Zombie'/113, l='MpServer', x=338.50, y=61.00, z=-191.50], ajr['Chicken'/114, l='MpServer', x=341.76, y=86.94, z=-186.55], amm['Spruce Sapling'/115, l='MpServer', x=339.13, y=92.00, z=-186.41], akm['Wolf'/116, l='MpServer', x=351.51, y=80.00, z=-177.43], amm['Bone'/117, l='MpServer', x=351.13, y=80.00, z=-177.26], ann['Skeleton'/118, l='MpServer', x=348.50, y=27.00, z=-167.50], amt['Creeper'/119, l='MpServer', x=351.50, y=23.00, z=-167.50], ann['Skeleton'/120, l='MpServer', x=345.50, y=23.00, z=-169.50], akm['Wolf'/121, l='MpServer', x=346.90, y=75.15, z=-151.18], ajr['Chicken'/648, l='MpServer', x=251.14, y=69.00, z=-272.48], ajr['Chicken'/649, l='MpServer', x=254.86, y=69.00, z=-277.75], ann['Skeleton'/657, l='MpServer', x=259.50, y=31.00, z=-256.50], anw['Zombie'/658, l='MpServer', x=265.49, y=58.00, z=-264.75], akm['Wolf'/148, l='MpServer', x=360.87, y=84.00, z=-210.30], ajr['Chicken'/149, l='MpServer', x=355.53, y=86.00, z=-192.88], amm['Egg'/150, l='MpServer', x=355.98, y=86.00, z=-192.58], ajr['Chicken'/151, l='MpServer', x=365.89, y=81.00, z=-188.46], anq['Spider'/664, l='MpServer', x=291.00, y=30.00, z=-280.01], ann['Skeleton'/152, l='MpServer', x=357.50, y=14.00, z=-164.50], amt['Creeper'/665, l='MpServer', x=290.50, y=30.00, z=-270.50], amt['Creeper'/153, l='MpServer', x=354.50, y=14.00, z=-168.50], amt['Creeper'/666, l='MpServer', x=296.50, y=28.00, z=-265.50], ajl['Bat'/154, l='MpServer', x=351.71, y=14.42, z=-166.41], amt['Creeper'/667, l='MpServer', x=298.50, y=28.00, z=-266.50], anw['Zombie'/155, l='MpServer', x=352.50, y=23.00, z=-167.50], anq['Spider'/156, l='MpServer', x=355.50, y=27.00, z=-166.50], ajl['Bat'/669, l='MpServer', x=310.35, y=38.03, z=-280.67], akm['Wolf'/157, l='MpServer', x=367.38, y=77.00, z=-172.48], ajl['Bat'/670, l='MpServer', x=310.50, y=38.08, z=-280.31], anw['Zombie'/158, l='MpServer', x=352.50, y=78.00, z=-164.30], ann['Skeleton'/159, l='MpServer', x=361.50, y=27.00, z=-153.50], ann['Skeleton'/160, l='MpServer', x=363.50, y=27.00, z=-151.50], anw['Zombie'/673, l='MpServer', x=319.50, y=34.00, z=-259.50], ann['Skeleton'/161, l='MpServer', x=359.70, y=27.00, z=-149.30], ann['Skeleton'/162, l='MpServer', x=363.50, y=27.00, z=-150.50], ctj['RoguePortal'/590, l='MpServer', x=310.47, y=94.00, z=-203.44], ajl['Bat'/229, l='MpServer', x=381.14, y=17.18, z=-244.92], ajl['Bat'/230, l='MpServer', x=378.49, y=32.19, z=-251.95], akf['Sheep'/231, l='MpServer', x=383.81, y=79.00, z=-210.50], anw['Zombie'/232, l='MpServer', x=373.50, y=40.00, z=-173.48], anw['Zombie'/233, l='MpServer', x=372.39, y=40.25, z=-172.07], ajr['Chicken'/234, l='MpServer', x=375.27, y=78.00, z=-171.90], ajr['Chicken'/235, l='MpServer', x=370.12, y=76.00, z=-162.51], apl['Minecart with Chest'/236, l='MpServer', x=382.50, y=25.06, z=-149.50]]
	Retry entities: 0 total; []
	Server brand: vanilla
	Server type: Integrated singleplayer server
Stacktrace:
	at crg.a(SourceFile:418)
	at cft.b(SourceFile:1815)
	at cft.a(SourceFile:403)
	at net.minecraft.client.main.Main.main(SourceFile:144)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

-- System Details --
Details:
	Minecraft Version: 1.13.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 281072976 bytes (268 MB) / 771751936 bytes (736 MB) up to 2147483648 bytes (2048 MB)
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Launched Version: 1.13.2-Impact_4.6
	LWJGL: 3.1.6 build 14
	OpenGL: GeForce GTX 760/PCIe/SSE2 GL version 4.6.0 NVIDIA 417.22, NVIDIA Corporation
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: Yes
	Is Modded: Very likely; Jar signature invalidated
	Type: Client (map_client.txt)
	Resource Packs: 
	Current Language: English (Australia)
	Profiler Position: N/A (disabled)
	CPU: 4x Intel(R) Core(TM) i5-4670K CPU @ 3.40GHz
```

</details>

**Comment by RoguePortal** [comment]
Deleting entire Impact folder in .minecraft solved the issue

**Comment by SuperOP535** [comment]
That's not a real solution

**Comment by LeafHacker** [comment]
This looks like a baritone issue @leijurv 

```
java.lang.NullPointerException: Ticking entity
	at java.util.Objects.requireNonNull(Objects.java:203)
	at baritone.p.a(Unknown Source)
	at baritone.ch$$Lambda$3456/1054571493.apply(Unknown Source)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:512)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:502)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:576)
	at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:255)
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:438)
	at baritone.ch.mineByName(Unknown Source)
	at baritone.api.process.IMineProcess.mineByName(Unknown Source)
```

**Comment by leijurv** [comment]
```    @Override
    public void mineByName(int quantity, String... blocks) {
        mine(quantity, blocks == null || blocks.length == 0 ? null : Arrays.stream(blocks).map(ChunkPacker::stringToBlockRequired).toArray(Block[]::new));
    }```

```    public static Block stringToBlockRequired(String name) {
        Block block = stringToBlockNullable(name);
        Objects.requireNonNull(block);
        return block;
    }```

This happens when Impact's Auto Mine utility module calls Baritone's mineByName with a String block name that doesn't exist. 

I saw this issue earlier, I think it was fixed by https://github.com/ImpactDevelopment/Impact/commit/6f54f37186769b43c37bdd05ff02df33a1749df4 perhaps? Or maybe this is a new issue. Hm, perhaps  https://github.com/ImpactDevelopment/Impact/blob/master/src/main/java/me/zero/clarinet/mod/impl/player/AutoMine.java#L104 ?

**Comment by leijurv** [comment]
Hmmmm, that wasn't it. I can't reproduce this...  I enabled Auto Mine and cycled through all the presets, and none of them crashed.

**Comment by leijurv** [comment]
This is fixed with latest hotfix (baritone 1.3.1)
